src/hex_registry.erl

```diff
@@ -49,13 +49,13 @@ encode_names(Names) ->
 
 %% @private
 decode_names(Payload, no_verify) ->
-    #{packages := Packages} = hex_pb_names:decode_msg(Payload, 'Names'),
-    {ok, Packages};
+    #{repository := Repository, packages := Packages} = hex_pb_names:decode_msg(Payload, 'Names'),
+    {ok, #{repository => Repository, packages => Packages}};
```

Comment:
hmm, actually should we just go with this?

```suggestion
    {ok, hex_pb_names:decode_msg(Payload, 'Names')}.
```

as long as it's safe, i.e. decode_msg doesn't return an error tuple or something?

in hindsight, it was a big mistake to return just the subset of the protobuf, just the packages field. With this change we're in sync with the protobuf which means if we ever add another field, we'll immediately start returning it here.

Reply:
Went ahead and made this change in [85e5964](https://github.com/hexpm/hex_core/pull/131/commits/85e59640cd48855bbf6ef0eb17d700d6fb24ac0a) — easy to revert if necessary. I also made the similar change to `decode_versions/2` and `decode_package/3`.